TestFoundation/TestOperationQueue.swift

```diff
@@ -168,34 +168,42 @@ class TestOperationQueue : XCTestCase {
 class AsyncOperation: Operation {
 
     private let queue = DispatchQueue(label: "async.operation.queue")
-    private let lock = NSLock()
+    private let lock = NSRecursiveLock()
```

Comment:
I changed the lock to `NSRecursiveLock()` because when `isFinished.set` is called and it takes the lock, the call to `didChangeValue` https://github.com/apple/swift-corelibs-foundation/blob/84ee193f9181838a366fe5f5e02d6345c7338c3a/Foundation/Operation.swift#L179  calls `isFinished.get` which would reaquire the lock. Making it recursive was the simplest fix. I didnt want to make any invasive changes in `Operation.swift`

What would be the benefit of the `DispatchQueue`? Would it not be too heavy an operation?